Error while checking build/O2/fullCI_slc9 for 70a163d02eb0964ff45097329a233ed4d7f18aaf at 2025-06-11 04:31:
```
## sw/BUILD/Rivet-latest/log
/sw/slc9_x86-64/YODA/yoda-2.1.0-3/include/YODA/highfive/H5Object.hpp:13:10: fatal error: H5Ipublic.h: No such file or directory
make[3]: *** [Makefile:563: libRivetCore_la-Run.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
/sw/slc9_x86-64/YODA/yoda-2.1.0-3/include/YODA/highfive/H5Object.hpp:13:10: fatal error: H5Ipublic.h: No such file or directory
/sw/slc9_x86-64/YODA/yoda-2.1.0-3/include/YODA/highfive/H5Object.hpp:13:10: fatal error: H5Ipublic.h: No such file or directory
make[3]: *** [Makefile:619: libRivetCore_la-AnalysisLoader.lo] Error 1
make[3]: *** [Makefile:633: libRivetCore_la-AnalysisHandler.lo] Error 1
/sw/slc9_x86-64/YODA/yoda-2.1.0-3/include/YODA/highfive/H5Object.hpp:13:10: fatal error: H5Ipublic.h: No such file or directory
make[3]: *** [Makefile:612: libRivetCore_la-Analysis.lo] Error 1
make[2]: *** [Makefile:659: all-recursive] Error 1
make[1]: *** [Makefile:567: all-recursive] Error 1
make: *** [Makefile:578: all-recursive] Error 1
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/14385/70a163d02eb0964ff45097329a233ed4d7f18aaf/build_O2_fullCI_slc9/pretty.html).

--- a/DataFormats/Detectors/EMCAL/include/DataFormatsEMCAL/CellLabel.h
+++ b/DataFormats/Detectors/EMCAL/include/DataFormatsEMCAL/CellLabel.h
@@ -15,6 +15,7 @@
 #include <cstddef>
 #include <cstdint>
 #include <gsl/span>
+#include <vector>
 
 namespace o2
 {
@@ -34,10 +35,15 @@ class CellLabel
  public:
   // CellLabel() = default;
 
-  /// \brief Constructor
+  /// \brief Constructor using std::vector by moving NOT copying
   /// \param labels list of mc labels
   /// \param amplitudeFractions list of amplitude fractions
-  CellLabel(const gsl::span<const int> labels, const gsl::span<const float> amplitudeFractions);
+  CellLabel(std::vector<int> labels, std::vector<float> amplitudeFractions);
+
+  /// \brief Constructor using gsl::span
+  /// \param labels list of mc labels
+  /// \param amplitudeFractions list of amplitude fractions
+  CellLabel(gsl::span<const int> labels, gsl::span<const float> amplitudeFractions);
 
   // ~CellLabel() = default;
   // CellLabel(const CellLabel& clus) = default;
@@ -52,21 +58,21 @@ class CellLabel
   int32_t GetLabel(size_t index) const { return mLabels[index]; }
 
   /// \brief Getter for labels
-  gsl::span<const int32_t> GetLabels() const { return mLabels; }
+  std::vector<int32_t> GetLabels() const { return mLabels; }
 
   /// \brief Getter for amplitude fraction
   /// \param index index which amplitude fraction to get
   float GetAmplitudeFraction(size_t index) const { return mAmplitudeFraction[index]; }
 
   /// \brief Getter for amplitude fractions
-  gsl::span<const float> GetAmplitudeFractions() const { return mAmplitudeFraction; }
+  std::vector<float> GetAmplitudeFractions() const { return mAmplitudeFraction; }
 
   /// \brief Getter for label with leading amplitude fraction
   int32_t GetLeadingMCLabel() const;
 
  protected:
-  gsl::span<const int32_t> mLabels;          ///< List of MC particles that generated the cluster, ordered in deposited energy.
-  gsl::span<const float> mAmplitudeFraction; ///< List of the fraction of the cell energy coming from a MC particle. Index aligns with mLabels!
+  std::vector<int32_t> mLabels;          ///< List of MC particles that generated the cluster, ordered in deposited energy.
+  std::vector<float> mAmplitudeFraction; ///< List of the fraction of the cell energy coming from a MC particle. Index aligns with mLabels!
 };
 
 } // namespace emcal

--- a/DataFormats/Detectors/EMCAL/src/CellLabel.cxx
+++ b/DataFormats/Detectors/EMCAL/src/CellLabel.cxx
@@ -16,10 +16,19 @@
 #include <cstddef>
 #include <cstdint>
 #include <gsl/span>
+#include <vector>
+#include <utility>
 
 using namespace o2::emcal;
 
-CellLabel::CellLabel(const gsl::span<const int> labels, const gsl::span<const float> amplitudeFractions) : mLabels(labels), mAmplitudeFraction(amplitudeFractions)
+CellLabel::CellLabel(std::vector<int> labels, std::vector<float> amplitudeFractions) : mLabels(std::move(labels)), mAmplitudeFraction(std::move(amplitudeFractions))
+{
+  if (labels.size() != amplitudeFractions.size()) {
+    LOG(error) << "Size of labels " << labels.size() << " does not match size of amplitude fraction " << amplitudeFractions.size() << " !";
+  }
+}
+
+CellLabel::CellLabel(gsl::span<const int> labels, gsl::span<const float> amplitudeFractions) : mLabels(labels.begin(), labels.end()), mAmplitudeFraction(amplitudeFractions.begin(), amplitudeFractions.end())
 {
   if (labels.size() != amplitudeFractions.size()) {
     LOG(error) << "Size of labels " << labels.size() << " does not match size of amplitude fraction " << amplitudeFractions.size() << " !";

